eth2/utils/eth2_key_derivation/src/derived_key.rs

```diff
@@ -144,25 +145,39 @@ fn parent_sk_to_lamport_pk(ikm: &[u8], index: u32) -> SecretHash {
 /// Equivalent to `IKM_to_lamport_SK` in EIP-2333.
 fn ikm_to_lamport_sk(salt: &[u8], ikm: &[u8]) -> LamportSecretKey {
     let prk = hkdf_extract(salt, ikm);
-    let okm = hkdf_expand(prk.as_bytes(), HASH_SIZE * LAMPORT_ARRAY_SIZE as usize);
+    let okm = hkdf_expand(prk, HASH_SIZE * LAMPORT_ARRAY_SIZE as usize);
     LamportSecretKey::from_bytes(okm.as_bytes())
 }
 
 /// Peforms a `HKDF-Extract` on the `ikm` (initial key material) based up on the `salt`.
 ///
 /// Defined in [RFC5869](https://tools.ietf.org/html/rfc5869).
-fn hkdf_extract(salt: &[u8], ikm: &[u8]) -> SecretHash {
+fn hkdf_extract(salt: &[u8], ikm: &[u8]) -> Prk {
+    Salt::new(HKDF_SHA256, salt).extract(ikm)
+    /*
     let mut prk = SecretHash::zero();
     crypto::hkdf::hkdf_extract(Sha256::new(), salt, ikm, prk.as_mut_bytes());
     prk
+    */
```

Comment:
Should the commented out parts be removed?

Reply:
Oh my bad! Didn't review the PR myself!